format/format.go

```diff
@@ -78,6 +78,20 @@ func Ast(x interface{}) (formatted []byte, err error) {
 		return false
 	})
 
+	// Rewrite the wildcards used in more than one place with their named
+	// values, they arent really wildcards anymore.
+	ast.WalkTerms(x, func(t *ast.Term) bool {
+		if v, ok := t.Value.(ast.Var); ok {
+			if v.IsWildcard() {
+				if newName, ok := wildcardNames[string(v)]; ok {
+					t.Value = ast.Var(newName)
+				}
+			}
+			return true
+		}
+		return false
+	})
+
```

Comment:
Would it be feasible to do this while walking the nodes ?

Reply:
🤔 Thats a good question. I suppose we could keep a reference to each term the first time we see them, so that we can rewrite it when we see another one. So yea, that should be pretty easy since we already have a map of "seen" ones, we can just have the value be the first term that we "saw".

Reply:
Just updated to do this, its a little more complex but does eliminate the additional walk over the AST 👍